gips/0058-indexing-fees.md

```diff
@@ -0,0 +1,218 @@
+
+**Sybil Resistant**
+Because consumers select Indexers, they will select Indexers that they know to be distinct. Therefore, Indexers have no incentive to split their identity.
+
+**Efficiency**
+It is documented that in a general class of large markets, posted prices are the most efficient pricing mechanism.
+
+**Simple**
+The new indexing fees mechanism mirrors standard market mechanisms: Sellers post a price and consumers select what to buy based on product quality and cost.
+
+## Automated Indexer Selection
+
+Although consumers may manually choose their own Indexers, those seeking a simpler experience could use an automatic Indexer selection algorithm. Such an algorithm may abstract away the need for consumers to screen individual Indexers. Instead, after a consumer provides their desired a) number of Indexers and b) sync speed preferences (high, medium, low), an automated Indexer selection algorithm can report the price and select the Indexers on behalf of the consumer.
+
+An example algorithm works as follows. First, Indexers are sorted based on their sync speed warranties and then categorized into high, medium or low sync speed (based on pre-defined thresholds). Indexer’s that are dominated, i.e. have a higher price for a lower sync speed warranty than some other Indexer — are removed from selection. The number of requested Indexers in the tier are selected sequentially.
+
+The first Indexer within the chosen sync speed tier is selected at uniform random. To provide robustness and guard against sybil attacks, the subsequent Indexers are selected at uniform random as long as their quality of service is not sufficiently correlated with previously selected Indexers. In other words, robustness and redundancy relies on Indexer performance being uncorrelated; this ensures, if one Indexer fails, there is no loss in service. By selecting Indexers that are uncorrelated, the automated algorithm ensures robustness. Furthermore, sybil
+identities are also likely to have a correlated quality of service, and so selecting Indexers with low correlation protects against sybil attacks.
+
+This example algorithm is admittedly crude and imprecise. However, it already gives consumers greater control over their quality of service than they have today and, thus, makes the protocol more efficient. Nevertheless, there is nothing preventing other agents from developing more sophisticated Indexer selection algorithms, incorporating both on-chain and off-chain data to improve consumer experience.
+
+## Frequently Asked Questions
+In this section, we address some of the frequently asked question that arise when replacing curation with indexing fees.
+
+**How do Indexers know which subgraphs to index?**
+They can be notified that they have been selected to index a subgraph at their posted price.
```

Comment:
Great note. We aren't prescribing it here, but will leave it open to the community (e.g. POI radio, Discord, Telegram, email). I suspect this will be a subject of a fair bit of conversation. I added a note to clarify.